internal/services/bot/bot_service_resource_base.go

```diff
@@ -306,6 +313,12 @@ func (br botBaseResource) readFunc() sdk.ResourceFunc {
 					streamingEndpointEnabled = *v
 				}
 				metadata.ResourceData.Set("streaming_endpoint_enabled", streamingEndpointEnabled)
+
+				iconUrl := ""
+				if v := props.IconURL; v != nil {
+					iconUrl = *v
+				}
+				metadata.ResourceData.Set("icon_url", iconUrl)
```

Comment:
fwiw this can become:

```suggestion
				metadata.ResourceData.Set("icon_url", pointer.To(props.IconURL))
```